tap_github/streams.py

```diff
@@ -151,57 +162,101 @@ def get_child_records(self,
+
+                        with singer.Transformer() as transformer:
 
-                else:
-                    # Write JSON response directly if it is a single record only.
-                    records['_sdc_repository'] = repo_path
-                    child_object.add_fields_at_1st_level(record = records, parent_record = parent_record)
+                            rec = transformer.transform(records, stream_catalog['schema'], metadata=metadata.to_map(stream_catalog['metadata']))
+                            if child_object.tap_stream_id in selected_stream_ids and records.get(child_object.replication_keys, start_date) >= child_bookmark_value :
 
+                                singer.write_record(child_object.tap_stream_id, rec, time_extracted=extraction_time)
+            elif child_object.no_path:
+                records = []
+                extraction_time = singer.utils.now()
+                if child_object.inherit_array_parent_fields: 
+                    for record in parent_record.get(child_object.inherit_array_parent_fields,[]):
+                        if col_name := child_object.custom_column_name: 
+                            records.append({col_name: record})
+                        else:
+                            records.append(record)
+                else: records.append({})
+                for record in records:
+                    for column, field in child_object.inherit_parent_fields:
+                        record[column] = parent_record.get(field)
+                    child_object.add_fields_at_1st_level(record = record, parent_record = parent_record)
                     with singer.Transformer() as transformer:
 
-                        rec = transformer.transform(records, stream_catalog['schema'], metadata=metadata.to_map(stream_catalog['metadata']))
-                        if child_object.tap_stream_id in selected_stream_ids and records.get(child_object.replication_keys, start_date) >= child_bookmark_value :
+                        rec = transformer.transform(record, stream_catalog['schema'], metadata=metadata.to_map(stream_catalog['metadata']))
 
+                        if child_object.tap_stream_id in selected_stream_ids:
                             singer.write_record(child_object.tap_stream_id, rec, time_extracted=extraction_time)
+                            counter.increment()
+
+                    # Loop thru each child and nested child in the parent and fetch all the child records.
+                    for nested_child in child_object.children:
+                        if nested_child in stream_to_sync:
+                            # Collect id of child record to pass in the API of its sub-child.
+                            child_id = tuple(record.get(key) for key in STREAMS[nested_child]().id_keys)
+                            if STREAMS[nested_child]().id_keys and not all(child_id): continue
```

Comment:
we are doing the `if STREAMS[nested_child]().id_keys` check after we access it in the previous line.
I guess it would be better to do a separate check for that and remove it from the second check